dbms/src/DataStreams/TiRemoteBlockInputStream.h

```diff
@@ -67,65 +67,25 @@ class TiRemoteBlockInputStream : public IProfilingBlockInputStream
 
     std::unique_ptr<CHBlockChunkDecodeAndSquash> decoder_ptr;
 
-    void initRemoteExecutionSummaries(tipb::SelectResponse & resp, size_t index)
+    void addRemoteExecutionSummaries(tipb::SelectResponse & resp, size_t index)
     {
-        for (const auto & execution_summary : resp.execution_summaries())
-        {
-            if (execution_summary.has_executor_id())
-            {
-                const auto & executor_id = execution_summary.executor_id();
-                execution_summaries[index][executor_id].time_processed_ns = execution_summary.time_processed_ns();
-                execution_summaries[index][executor_id].num_produced_rows = execution_summary.num_produced_rows();
-                execution_summaries[index][executor_id].num_iterations = execution_summary.num_iterations();
-                execution_summaries[index][executor_id].concurrency = execution_summary.concurrency();
-            }
-        }
-        execution_summaries_inited[index].store(true);
-    }
-
-    void addRemoteExecutionSummaries(tipb::SelectResponse & resp, size_t index, bool is_streaming_call)
-    {
-        if (resp.execution_summaries_size() == 0)
+        if (unlikely(resp.execution_summaries_size() == 0))
             return;
 
-        if (!execution_summaries_inited[index].load())
-        {
-            initRemoteExecutionSummaries(resp, index);
-            return;
-        }
-        auto & execution_summaries_map = execution_summaries[index];
+        // There will only be one summary packet for one index.
```

Comment:
This is not true for remote read :(, maybe we should seperate the execution summary part for `CoprocessorReader` and `ExchangeReceiver`.

Reply:
O.....yes...CoprocessReader will send multiple cop tasks....